iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/WorkdirInsteadCdCheck.java

```diff
@@ -61,6 +62,13 @@ private static void check(CheckContext ctx, CommandInstruction commandInstructio
     }
   }
 
+  private static void checkRunInstruction(CheckContext ctx, RunInstruction runInstruction) {
```

Comment:
WDYT about introducing a high-order function in CheckUtils like `ignoringHeredoc(BiConsumer<CheckContext, RunInstruction>)` and using it directly in `init.register`? I think this could unify all the places where we will ignore heredocs and make it easier to discover them once we improve heredoc support.

Reply:
Really nice idea!